c/tskit/trees.c

```diff
@@ -315,13 +315,31 @@ tsk_treeseq_init_migrations(tsk_treeseq_t *self)
     tsk_size_t num_migrations = self->tables->migrations.num_rows;
     const double *restrict left = self->tables->migrations.left;
     const double *restrict right = self->tables->migrations.right;
+    const double *restrict time = self->tables->migrations.time;
     bool discrete_breakpoints = true;
+    bool discrete_times = true;
 
     for (j = 0; j < num_migrations; j++) {
         discrete_breakpoints
             = discrete_breakpoints && is_discrete(left[j]) && is_discrete(right[j]);
+        discrete_times = discrete_times && is_discrete(time[j]);
     }
     self->discrete_genome = self->discrete_genome && discrete_breakpoints;
+    self->discrete_time = self->discrete_time && discrete_times;
+}
+
+static void
+tsk_treeseq_init_mutations(tsk_treeseq_t *self)
+{
+    tsk_size_t j;
+    tsk_size_t num_mutations = self->tables->mutations.num_rows;
+    const double *restrict time = self->tables->mutations.time;
+    bool discrete_times = true;
+
+    for (j = 0; j < num_mutations; j++) {
+        discrete_times = discrete_times && is_discrete(time[j]);
```

Comment:
I think it'd be clearer if we put the special case for ``is_unknown_time`` in here - it's more obvious that we're making an exception for it, and we also avoid calling ``tsk_is_unknown_time`` for things like genome positions, which can't be unknown.